lib/wallet/mod.rs

```diff
@@ -566,40 +567,73 @@ impl Task {
     /// see https://docs.rs/tokio/latest/tokio/task/fn.block_in_place.html
     /// A shutdown signal is used to ensure that the task stops after
     /// the blocking task completes.
-    async fn sync_task(wallet: Arc<WalletInner>, mut shutdown_rx: oneshot::Receiver<()>) {
+    async fn sync_task(
+        wallet: Arc<WalletInner>,
+        start_rx: oneshot::Receiver<()>,
+        mut shutdown_rx: oneshot::Receiver<()>,
+    ) {
+        use futures::future::{select, Either};
         const SYNC_INTERVAL: Duration = Duration::from_secs(15);
-        tracing::debug!("wallet sync task: starting with interval {SYNC_INTERVAL:?}",);
-
-        let mut interval = interval(SYNC_INTERVAL);
+        tracing::debug!(
+            interval = %humantime::format_duration(SYNC_INTERVAL),
+            "wallet sync task: starting"
+        );
+        shutdown_rx = match select(start_rx, shutdown_rx).await {
```

Comment:
I had to stare at this for a bit before understanding what's happening. IMO a comment like this would be great. 

`Wait for the start signal to be sent before starting the sync. Take care to also listen for the shutdown signal or channel closures, which might happen while waiting for start`